lib/engine/game/g_1880/step/buy_train.rb

```diff
@@ -55,7 +58,14 @@ def discard_trains?
           def discard_all_trains(train_name)
             @game.log << "#{train_name} has not been bought for a full operating order, "\
                          "removing all remaining #{train_name} trains"
-            @game.depot.export_all!(train_name, silent: true)
+            return @game.depot.export_all!(train_name, silent: true) unless special_edge_case_4t_rocket?(train_name)
+
+            @game.depot.export!(train_name, silent: true)
+            @round.special_edge_case_4t_rocket = true
+          end
```

Comment:
only export one if we're in 4t rocket, edge case territory.
The assign step will kick in, and prompt the owner of the rocket to pick corp to gain the 4